sumomongodbatlascollector/api.py

```diff
@@ -24,11 +25,21 @@ def __init__(self, kvstore, config):
     def get_window(self, last_time_epoch):
         start_time_epoch = last_time_epoch + self.MOVING_WINDOW_DELTA
         end_time_epoch = get_current_timestamp() - self.collection_config['END_TIME_EPOCH_OFFSET_SECONDS']
-        if end_time_epoch < start_time_epoch:
+        while not (end_time_epoch - start_time_epoch > 60):
```

Comment:
Why do we need to do this (to reduce API calls, etc.)? Add a comment here.

Reply:
but isn't it clear it from below comment in line 29 that I am doing this because endtime becomes lesser than starttime

Reply:
No, not that kind of comment. I am looking for comments on the collection window logic, like the window, how we move things forward.

Reply:
gotcha window moving logic is in design doc let me add the link, why we sleep for 60 (let me put this in variable it should MIN_REQUEST_WINDOW_LENGTH) is good point because previously it was 2 sec I have observed that the collector was even requesting data for very small intervals since our window is (last event time to current time).